xbmc/addons/AddonManager.cpp

```diff
@@ -1015,24 +1024,57 @@ bool CAddonMgr::PlatformSupportsAddon(const cp_plugin_info_t *plugin)
 #elif defined(TARGET_WINDOWS_DESKTOP)
     "windx",
     "windows",
+#if defined(_M_IX86)
+    "windows-i686",
+#elif defined(_M_AMD64)
+    "windows-x86_64",
+#else
+#error no architecture dependant platform tag
+#endif
 #elif defined(TARGET_WINDOWS_STORE)
     "windowsstore",
 #elif defined(TARGET_DARWIN_IOS)
     "ios",
+#if defined(__ARM_ARCH_7A__)
+    "ios-armv7",
+#elif defined(__aarch64__)
+    "ios-aarch64",
+#else
+#warning no architecture dependant platform tag
+#endif
 #elif defined(TARGET_DARWIN_OSX)
     "osx",
 #if defined(__x86_64__)
     "osx64",
-#else
+    "osx-x86_64",
+#elif defined(__i686__)
+    "osx-i686",
     "osx32",
+#else
+#warning no architecture dependant platform tag
 #endif
 #endif
   };
 
-  return std::find_first_of(platforms.begin(), platforms.end(),
-      supportedPlatforms.begin(), supportedPlatforms.end()) != platforms.end();
+  auto firstMatch = std::find_first_of(platforms.begin(), platforms.end(), supportedPlatforms.begin(), supportedPlatforms.end());
+  if (firstMatch != platforms.end())
+    platformMatch = *firstMatch;
+  return firstMatch != platforms.end();
 }
 
+bool CAddonMgr::PlatformDependantAddon(const cp_plugin_info_t *plugin)
+{
+  auto *metadata = CServiceBroker::GetAddonMgr().GetExtension(plugin, "xbmc.addon.metadata");
+  if (!metadata)
+    metadata = CServiceBroker::GetAddonMgr().GetExtension(plugin, "kodi.addon.metadata");
+
+  // if platform dependant is not specified, assume not platformdependant
+  if (!metadata)
+    return false;
+
+  return CServiceBroker::GetAddonMgr().GetExtValue(metadata->configuration, "platformdependant") == "true";
+}
```

Comment:
Please follow the existing structure and keep metadata parsing in the `Factory` method. Even if you just need it for the path right now, there is no need to throw this info away.